gateway/load_balancer_test.go

```diff
@@ -0,0 +1,148 @@
+package gateway
+
+import (
+	"io"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+)
+
+type mockDNSResolver struct {
+	IPs []net.IP
+	Err error
+}
+
+func (m mockDNSResolver) LookupIP(hostname string) ([]net.IP, error) {
+	return m.IPs, m.Err
```

Comment:
Unfortunately this is not how it works. The DNS returns IPs in different order every time. 

for example:

**First try**

```
$ nslookup cortexmetrics.io
Server:		10.128.4.247
Address:	10.128.4.247#53

Non-authoritative answer:
Name:	cortexmetrics.io
Address: 172.67.146.72
Name:	cortexmetrics.io
Address: 104.21.73.166
```
**Second try**

```
$ nslookup cortexmetrics.io
Server:		10.128.4.247
Address:	10.128.4.247#53

Non-authoritative answer:
Name:	cortexmetrics.io
Address: 104.21.73.166
Name:	cortexmetrics.io
Address: 172.67.146.72
```

I think if you randomize the order here, it can get pretty similar, though